lua/lazyvim/plugins/extras/lang/java.lua

```diff
@@ -1,125 +1,151 @@
+          "-data",
+          workspace_dir,
+        },
+        root_dir = require("jdtls.setup").find_root({
+          ".git",
+          "mvnw",
+          "gradlew",
+          "pom.xml",
+          "build.gradle",
+          "build.gradle.kts",
+          "settings.gradle",
+          "settings.gradle.kts",
+          "build.xml",
+          ".project",
+        }),
+        settings = {
+          java = {},
+        },
+        -- silent setup
+        handlers = {
+          ["$/progress"] = function() end,
+          ["language/status"] = function() end,
+        },
+        init_options = {
+          bundles = {
+            vim.fn.expand("$MASON/share/java-debug-adapter/com.microsoft.java.debug.plugin.jar"),
+            (table.unpack or unpack)(vim.split(vim.fn.glob("$MASON/share/java-test/*.jar"), "\n", {})),
+          },
+        },
+        filetypes = { "java" },
+        capabilities = require("cmp_nvim_lsp").default_capabilities(), -- enable CMP capabilities
+      }
+
+      -- attach every Java buffer
+      vim.api.nvim_create_autocmd("FileType", {
+        pattern = "java",
+        callback = function()
+          require("jdtls").start_or_attach(config)
+        end,
+      })
+
+      -- setup keymap and DAP after the LSP is fully attached
+      -- https://github.com/mfussenegger/nvim-jdtls#nvim-dap-configuration
+      -- https://neovim.io/doc/user/lsp.html#LspAttach
+      vim.api.nvim_create_autocmd("LspAttach", {
+        pattern = "*.java",
+        callback = function(args)
+          require("which-key").register({
+            ["<leader>cx"] = { name = "+extract" },
+            ["<leader>cxv"] = { require("jdtls").extract_variable_all, "Extract Variable" },
+            ["<leader>cxc"] = { require("jdtls").extract_constant, "Extract Constant" },
+            ["gs"] = { require("jdtls").super_implementation, "Goto Super" },
+            ["gS"] = { require("jdtls.tests").goto_subjects, "Goto Subjects" },
+            ["<leader>co"] = { require("jdtls").organize_imports, "Organize Imports" },
+          }, { mode = "n", buffer = args.buf })
+          require("which-key").register({
+            ["<leader>c"] = { name = "+code" },
+            ["<leader>cx"] = { name = "+extract" },
+            ["<leader>cxm"] = { [[<ESC><CMD>lua require('jdtls').extract_method(true)<CR>]], "Extract Method" },
+            ["<leader>cxv"] = { [[<ESC><CMD>lua require('jdtls').extract_variable_all(true)<CR>]], "Extract Variable" },
+            ["<leader>cxc"] = { [[<ESC><CMD>lua require('jdtls').extract_constant(true)<CR>]], "Extract Constant" },
+          }, { mode = "v", buffer = args.buf })
+
+          require("jdtls.dap").setup_dap()
+          require("jdtls.dap").setup_dap_main_class_configs()
+
+          require("which-key").register({
+            ["<leader>t"] = { name = "+test" },
+            ["<leader>tT"] = { require("jdtls.dap").test_class, "Run All Test" },
```

Comment:
not sure if this only registers the keys for *.java buffers, or in general. If the latter it might break test integrations done via neotest for other languages in the same project (e.g. javascript/typescript via neotest-jest)